Include/internal/pycore_runtime_init.h

```diff
@@ -94,6 +93,29 @@ extern "C" {
         _PyBytes_SIMPLE_INIT(CH, 1) \
     }
 
+#define _PyASCIIObject_INIT(LITERAL) \
+    { \
+        ._ascii = { \
+            .ob_base = { \
+                .ob_refcnt = 999999999, \
+                .ob_type = &PyUnicode_Type, \
+            }, \
+            .length = _Py_STRING_LENGTH(LITERAL) - 1, \
+            .hash = -1, \
+            .state = { \
+                .kind = 1, \
+                .compact = 1, \
+                .ascii = 1, \
+                .ready = 1, \
+            }, \
+        }, \
+        ._data = LITERAL, \
+    }
+#define INIT_STR(NAME, LITERAL) \
+    .spamspamspam_ ## NAME = _PyASCIIObject_INIT(LITERAL)
+#define INIT_ID(NAME) \
+    .spamspamspam_ ## NAME = _PyASCIIObject_INIT(#NAME)
+
 
 /* The following is auto-generated by Tools/scripts/generate_global_objects.py. */
 #define _Py_global_objects_INIT { \
```

Comment:
It is used in 2 places (pystate.c and pylifecycle.c).  Otherwise I agree.  Also, I am not opposed to getting rid of the optimization in pystate.c if you think it isn't worth it.  Then we could drop the giant macro.

Reply:
What is the "optimization in pystate.c"?

Reply:
It is only used once.
```
$ git grep _Py_global_objects_INIT
Include/internal/pycore_runtime_init.h:        .global_objects = _Py_global_objects_INIT, \
Include/internal/pycore_runtime_init.h:#define _Py_global_objects_INIT { \
Tools/scripts/generate_global_objects.py:        with printer.block('#define _Py_global_objects_INIT', continuation=True):
```

Reply:
Sorry, I thought you were talking about `_PyRuntimeState_INIT()`.  Certainly I could fold `_Py_global_objects_INIT()` into `_PyRuntimeState_INIT()`.  However, only `_Py_global_objects_INIT()` is generated code.  Generating only a portion of the initializer would be messy.